### DISCUSSION TITLE
Massive geojson points file - memory issue

### DISCUSSION THREAD

**Discussion by Jul-Git** [TerriaJS/terriajs#6273] [Q&A]
Hello everyone,

I am trying to load a huge (in my point of view) geojson point file (over 150000 points with descriptions). On some PC, with not enough RAM, it is not loading at all.
Is there a workaround to load such massive dataset into TerriaMap ?

**Comment by nf-s** [comment]
Hi @Jul-Git 

We support mapbox vector tiles. So you can use something like [`tippecanoe`](https://github.com/mapbox/tippecanoe) to tile and generalise your geojson points file/

They have an example with large point dataset here - https://github.com/mapbox/tippecanoe#large-point-dataset-gps-bus-locations-for-visualization-at-all-zoom-levels

You can then add a `MapboxVectorTileCatalogItem` like so:

```json
{
      "name": "Points",
      "type": "mvt",
      "url": "https://some-domain/some-path/{z}/{x}/{y}.vector.pbf",
      "layer": "layer-name"
      ...
},
```

See [`MapboxVectorTileCatalogItem` docs](https://docs.terria.io/guide/connecting-to-data/catalog-type-details/mvt/)
